desktop/core/src/desktop/js/apps/notebook2/components/aceEditor/autocomplete/AutocompleteResults.ts

```diff
@@ -726,7 +728,7 @@ class AutocompleteResults {
 
     AutocompleteResults.mergeColumns(columnSuggestions);
 
-    if (this.dialect() === DIALECT.hive && /[^.]$/.test(this.editor.getTextBeforeCursor())) {
+    if (this.dialect() === HIVE_DIALECT && /[^.]$/.test(this.editor.getTextBeforeCursor())) {
```

Comment:
Here those seems to be dialect specific corner cases? (e.g. nested type so fair to keep hardcoded)

Reply:
These are for built in virtual columns in hive, I think it'd be better to move them to the parser at some point.